crates/sui-adapter/src/adapter.rs

```diff
@@ -644,6 +647,12 @@ pub fn resolve_and_type_check(
             })
         }
     };
+    // Check for script visibility, but ignore for genesis
```

Comment:
Why are we ignoring it for genesis?

Reply:
Lol @lxfind do you have a good comment here?

I can at least say for now something like "genesis calls private functions, which is helpful for ensuring the functions are not called again later. "
?

Reply:
It would be good to have a bit context in the comment on this.

Reply:
Added, let me know it looks good before landing